character-sfx.js

```diff
@@ -62,22 +72,29 @@ const _getActionFrameIndex = (f, frameTimes) => {
   return i;
 };
 
-class CharacterSfx {
+class CharacterSfx extends EventTarget{
   constructor(player) {
+    super();
     this.player = player;
 
     this.lastJumpState = false;
     this.lastStepped = [false, false];
     this.lastWalkTime = 0;
+
+   
+    
+    this.lastSwordComboName = null;

```

Comment:
There are now so many random state keys on this class that it needs to be broken up further.